src/pages/search.js

```diff
@@ -40,12 +52,87 @@ export default withSearch(({ search }) => {
     ? ` : ${query} (${totalHits} result${totalHits === 1 ? '' : 's'})`
     : ''
 
-  let searchEvent
+  const getSlug = item => {
+    if (!item.publishDate) {
+      return `${item.slug}`
+    }
+    return `/blog/${item.slug}`
+  }
+
+  const goToResult = selectedItem => {
+    const allResults = [
+      ...results.states.hits,
+      ...results.blogPosts.hits,
+      ...results.pages.hits,
+    ]
+
+    const item = allResults.find(result => {
+      return result.name === selectedItem || result.title === selectedItem
+    })
+
+    if (item && typeof window !== 'undefined') {
+      window.location = getSlug(item)
```

Comment:
@jasonlcrane check out `withLocation` component, it passes `location` and `navigate` if I remember well.
You could add `navigate` to the component's props `export default withSearch(({ search, navigate }) => {` so you can then use it to go to the destination while retaining history.
(https://reach.tech/router/api/navigate)

Reply:
Thanks, good call. Updated.